models/recipe.js

```diff
@@ -0,0 +1,47 @@
+const mongoose = require("mongoose");
+
+const recipeSchema = mongoose.Schema(
+  {
+    title: {
+      type: String,
+      required: true,
+    },
+    picturePath: {
+      type: String,
+      default: "",
+    },
+    servings: Number,
+    steps: [
+      {
+        stepName: {type: String},
+        step: [{type: String}],
+      },
+    ],
+    ingredients: [
+      {
+        ingredientsFor: String,
+        allIngredients: [
```

Comment:
what are these for?

I think it could just be 
```
ingredients: [
  {
      ingredient: {type: String, required: true},
      unit: {type: String},
      amount: {type: Number, required: true},
    },
]
```

Reply:
Yes! I had to do this because some recipes have ingredients sections 
ie A separate ingredient section for the base, icing, and filling. 

Sometimes the ingredients would overlap, this is why I did this.